.evergreen/install-dependencies.sh

```diff
@@ -60,4 +62,5 @@ registry=https://registry.npmjs.org
 EOT
 
 # install node dependencies
-npm install
\ No newline at end of file
+npm install
```

Comment:
hm, I don't think this will cut it. You would need to do something like:
```
nvm install --no-progress --lts=dubnium
npm install # implicitly does an `npm build:ts` or whatever
git clean -dfx
nvm install --no-progress ${NODE_VERSION}
npm install
```

Also, can you please document this more carefully, this actually somewhat tricky how it's working right now. I suspect this is working because of the `prepare` script checking for the presence of the `lib` folder and bailing from compilation otherwise. It's not clear from this source that `prepare` is being used this way, nor is it even clear that `npm install` is doing the initial compilation.

Reply:
I added a comment about npm install doing the compilation with prepare.
If you run `git clean` you will delete the compiled lib folder and the downloaded dependencies. The prepare script doesn't bail if there's a lib folder, it checks for a src directory which is only present if you cloned the repo. It does also bail if the node version is too old, but it will warn you if you lack a lib folder, and even further warn you that you can't compile the lib on older versions of node. Here's the prepare script:
```js
#! /usr/bin/env node
var cp = require('child_process');
var fs = require('fs');

var nodeMajorVersion = +process.version.match(/^v(\d+)\.\d+/)[1];

if (fs.existsSync('src') && nodeMajorVersion >= 10) {
  cp.spawnSync('npm', ['run', 'build'], { stdio: 'inherit', shell: true });
} else {
  if (!fs.existsSync('lib')) {
    console.warn('BSON: No compiled javascript present, the library is not installed correctly.');
    if (nodeMajorVersion < 10) {
      console.warn(
        'This library can only be compiled in nodejs version 10 or later, currently running: ' +
          nodeMajorVersion
      );
    }
  }
}
```